File: app/schemas/review.py
"""
Schemas Pydantic para Review (Reseñas)
"""
from pydantic import BaseModel, Field
from pydantic import ConfigDict
from typing import Optional
from datetime import datetime
from uuid import UUID


class ReviewBase(BaseModel):
    rating: int = Field(..., ge=1, le=5, description="Calificación del libro (1-5)")
    comment: Optional[str] = Field(None, description="Comentario opcional sobre el libro")


class ReviewCreate(ReviewBase):
    book_id: UUID = Field(..., description="ID del libro reseñado")
    group_id: Optional[UUID] = Field(None, description="ID del grupo asociado (opcional)")


class ReviewUpdate(BaseModel):
    rating: Optional[int] = Field(None, ge=1, le=5)
    comment: Optional[str] = None


class ReviewInDB(ReviewBase):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    book_id: UUID
    user_id: UUID
    group_id: Optional[UUID] = None
    created_at: datetime
    updated_at: Optional[datetime] = None


class ReviewResponse(ReviewInDB):
    # Relaciones opcionales para respuestas enriquecidas
    book_title: Optional[str] = None
    user_username: Optional[str] = None
    group_name: Optional[str] = None

File: frontend/lib/types/review.ts
// Tipos para el sistema de reseñas

export interface Review {
  id: string;
  rating: number; // 1-5
  comment?: string;
  book_id: string;
  user_id: string;
  group_id?: string;
  created_at: string;
  updated_at?: string;
  
  // Datos enriquecidos (opcionales)
  book_title?: string;
  user_username?: string;
  group_name?: string;
}

export interface ReviewCreate {
  rating: number; // 1-5
  comment?: string;
  book_id: string;
  group_id?: string;
}

export interface ReviewUpdate {
  rating?: number; // 1-5
  comment?: string;
}

export interface ReviewFilters {
  book_id?: string;
  user_id?: string;
  group_id?: string;
  limit?: number;
  offset?: number;
}

export interface ReviewStats {
  total_reviews: number;
  average_rating: number;
  rating_distribution: {
    [key: number]: number; // { 1: 5, 2: 3, 3: 10, 4: 15, 5: 20 }
  };
}
